xds/src/test/java/io/grpc/xds/LookasideLbTest.java

```diff
@@ -125,111 +128,50 @@ public void uncaughtException(Thread t, Throwable e) {
           throw new AssertionError(e);
         }
       });
-  private final StreamRecorder<DiscoveryRequest> streamRecorder = StreamRecorder.create();
-
-  private final DiscoveryResponse edsResponse =
-      DiscoveryResponse.newBuilder()
-          .addResources(Any.pack(ClusterLoadAssignment.getDefaultInstance()))
-          .setTypeUrl("type.googleapis.com/envoy.api.v2.ClusterLoadAssignment")
-          .build();
-  private final Deque<Helper> helpers = new ArrayDeque<>();
-  private final Deque<LocalityStore> localityStores = new ArrayDeque<>();
-  private final Deque<LoadReportClient> loadReportClients = new ArrayDeque<>();
   private final FakeClock fakeClock = new FakeClock();
 
+  // Monitor responseObservers on the server that the bootstrapChannel connects to.
+  private final ArrayDeque<StreamObserver<DiscoveryResponse>> responseObservers =
+      new ArrayDeque<>();
+  // Monitor endpointWatcher added to xdsClientFromResolvedAddresses.
+  private final Map<String, EndpointWatcher> endpointWatchers = new HashMap<>();
+
+  private final LoadBalancerRegistry lbRegistry = new LoadBalancerRegistry();
+
+  // Child helpers keyed by locality names.
+  private final Map<String, Helper> childHelpers = new HashMap<>();
+  // Child balancers keyed by locality names.
+  private final Map<String, LoadBalancer> childBalancers = new HashMap<>();
+  // LocalityStores created by fake localityStoreFactory.
+  private final ArrayDeque<LocalityStore> localityStores = new ArrayDeque<>();
+  // LoadReportClients created by xdsClientFromResolvedAddresses.
+  private final ArrayDeque<LoadReportClient> loadReportClients = new ArrayDeque<>();
+
   @Mock
   private Helper helper;
   @Mock
   private EndpointUpdateCallback edsUpdateCallback;
-  @Mock
-  private Bootstrapper bootstrapper;
   @Captor
-  private ArgumentCaptor<ImmutableMap<Locality, LocalityLbEndpoints>>
-      localityEndpointsMappingCaptor;
+  ArgumentCaptor<ConnectivityState> connectivityStateCaptor;
+  @Captor
+  ArgumentCaptor<SubchannelPicker> pickerCaptor;
 
-  private ManagedChannel channel;
-  private ManagedChannel channel2;
-  private StreamObserver<DiscoveryResponse> serverResponseWriter;
   private LoadBalancer lookasideLb;
-  private ResolvedAddresses defaultResolvedAddress;
+  private boolean withBootstrap;
```

Comment:
Using boolean flags to switch different test cases is a bad style. If the test setup for different use cases are very different, it's better to split into two test classes.

Reply:
Every test case has a different setup (not talking about the global `@Before` setup). The flags is just a state of the test case, it's not switching to different test case based on the flog.

Reply:
Okay, using one flag.